### PR TITLE
feat: separate starship installation by using fixed image

### PR DESCRIPTION
**Pull request by Kaiyang-Chen**
Signed-off-by: Kaiyang Chen <kaiyang-chen@sjtu.edu.cn>

### CODE REVIEW

File: base-images/envd-starship/envd-starship.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM ubuntu:20.04
+RUN apt-get update && apt-get install -y apt-utils
+RUN apt-get install -y --no-install-recommends --no-install-suggests --fix-missing curl ca-certificates
```
Comment (kemingy) [inline base-images/envd-starship/envd-starship.Dockerfile:3]: You can use a curl image here.

File: base-images/envd-starship/envd-starship.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+RUN curl --proto '=https' --tlsv1.2 -sSf https://starship.rs/install.sh | sh -s -- -y
```
Comment (VoVAllen) [inline base-images/envd-starship/envd-starship.Dockerfile:4]: Better to use multi-stage build here if only one binary is needed here. For example, https://github.com/tensorchord/Horust/blob/master/Dockerfile#L12

**Review by VoVAllen — COMMENTED** [review]
And please also add an github action to upload the image like https://github.com/tensorchord/Horust/blob/master/.github/workflows/release.yml

File: base-images/envd-starship/envd-starship.Dockerfile (hunk 1 above)
Comment (Kaiyang-Chen) [reply base-images/envd-starship/envd-starship.Dockerfile]: make sense, PTAL

File: base-images/envd-starship/envd-starship.Dockerfile (hunk 2 above)
Comment (Kaiyang-Chen) [reply base-images/envd-starship/envd-starship.Dockerfile]: make sense, PTAL

File: base-images/envd-starship/envd-starship.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FROM curlimages/curl:7.87.0 as builder
+USER root
+RUN curl --proto '=https' --tlsv1.2 -sSf https://starship.rs/install.sh | sh -s -- -y
+
+FROM ubuntu:20.04 as prod
```
Comment (kemingy) [inline base-images/envd-starship/envd-starship.Dockerfile:5]: You can use scratch.

**Review by kemingy — COMMENTED** [review]
BTW, shall we also change for v0?

**Review by kemingy — DISMISSED** [review]
LGTM. Shall we merge the CI part to release the image or do it manually before merging the rest of this PR?
@gaocegege

File: .github/workflows/envd-starship.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERIO_USERNAME }}
+          password: ${{ secrets.DOCKERIO_TOKEN }}
+      - name: Docker Setup QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: build image
+        run: |
```
Comment (VoVAllen) [inline .github/workflows/envd-starship.yaml:30]: Use build and push action instead. Example: https://github.com/tensorchord/envd-server/blob/main/.github/workflows/release.yml#L137-L146

Comment (Kaiyang-Chen) [reply .github/workflows/envd-starship.yaml]: changed and tested. PTAL.

**Comment by gaocegege** [comment]
> LGTM. Shall we merge the CI part to release the image or do it manually before merging the rest of this PR? @gaocegege

I think we can release a new version before the PR, then merge it to let CI release a new image. WDYT

**Comment by kemingy** [comment]
> > LGTM. Shall we merge the CI part to release the image or do it manually before merging the rest of this PR? @gaocegege
> 
> I think we can release a new version before the PR, then merge it to let CI release a new image. WDYT

LGTM

**Comment by VoVAllen** [comment]
https://github.com/all-contributors please add @Kaiyang-Chen for the code contribution.

**Comment by VoVAllen** [comment]
@all-contributors please add @Kaiyang-Chen for the code contribution.

**Pull request merged**